### PR TITLE
SDK-178 Java BDK: Create test coverage for sdk.symphony.MessageClientImpl

### PR DESCRIPTION
**Pull request by Haiddi-Avencod**
### Ticket
[SDK-178](https://perzoinc.atlassian.net/browse/SDK-178?atlOrigin=eyJpIjoiMmZkNzFjNGY0ZjUzNDI5Y2JjYTg1NGFjNzRiNjMxNTMiLCJwIjoiaiJ9)

### Coverage
coverage_178

### CODE REVIEW

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    this.messageClientImpl.sendMessage(streamId, message, jsonData, messageAttachmentFiles);
+  }
+
+  @Test
+  public void testDownloadMessageAttachments() throws SymphonyClientException {
+    final MessageEvent messageEvent = this.initMessageEvent();
+    final byte[] fileContent1 = new byte[] {(byte) 6789, (byte) 47, (byte) 567};
+    final byte[] fileContent2 = new byte[] {(byte) 990, (byte) 467, (byte) 5654};
+    final byte[] fileContent3 = new byte[] {(byte) 687, (byte) 479, (byte) 574};
+    final List<FileAttachment> fileAttachmentList = this.initFileAttachmentsList(fileContent1, fileContent2, fileContent3);
```
Comment (symphony-elias) [inline symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: Why not reuse `this.initMessageAttachmentFiles()` ?

**Review by symphony-elias — COMMENTED** [review]
Please have a look at the comments.

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java (hunk 1 above)
Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: I don't have to init a `List<MessageAttachmentFile>`, but I have to init a `List<FileAttachment>` here. I don't see how you want me to do so

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: Maybe I create a method `initFileAttachmentList` ?

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: Maybe I put this code : 
```
    final byte[] fileContent1 = new byte[] {(byte) 6789, (byte) 47, (byte) 567};
    final byte[] fileContent2 = new byte[] {(byte) 990, (byte) 467, (byte) 5654};
    final byte[] fileContent3 = new byte[] {(byte) 687, (byte) 479, (byte) 574};
```
in initFileAttachmentsList

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: No, i can't do this

Comment (symphony-elias) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/MessageClientImplTest.java:107]: Sorry, types are different. This is so misleading.

**Review by symphony-elias — APPROVED** [review]
Looks good 👍

**Pull request merged**